git-config/src/file/init/comfort.rs

```diff
@@ -95,6 +96,20 @@ impl File<'static> {
             (local, path)
         };
 
+        let worktree = match local.boolean("extensions", None, "worktreeConfig") {
+            Some(Ok(worktree_config)) => worktree_config.then(|| {
```

Comment:
For reference in `git-repository`: We won't ignore errors there, instead there is `lenient` config toggle which allows to ignore errors like here only if it's set. Otherwise, it's an error to not be able to parse this a boolean. Handling `lenient` is very convenient by now.